### ISSUE TITLE
All egs++ shapes should be converted to shared libraries

### ISSUE THREAD

**Issue by rtownson** [nrc-cnrc/EGSnrc#1213]
A few simple shapes are defined with `type=` instead `library=` and this can be confusing for users. The shapes might as well be made available as shared libraries just like the rest.

**Comment by ftessier** [comment]
Agreed. This is always a nuisance, even for experienced EGSnrc users! We can check whether this affects efficiency, but I presume it won't have a significant impact. Is there any mention in the docs why it was done this way in the first place?
